exchange_machine.lua

```diff
@@ -225,32 +215,28 @@ minetest.after(0, function()
 		inv:add_item("main", ItemStack(itemstring))
 	end
 
-	selectable_inventory_size = #selectable_list
+	pagemax = math.ceil((#selectable_list - 1) / (8 * 4))
 end)
 
 
 minetest.register_on_joinplayer(function(player)
-	local big_inv = minetest.get_inventory({type="detached", name="global_exchange"})
-	local p_list = "p_" .. player:get_player_name()
-
-	big_inv:set_size(p_list, 1)
+	minetest.get_inventory({
+		type="detached",
+		name="global_exchange"
+	}):set_size("p_" .. player:get_player_name(), 1)
 end)
-	
+
 
 local select_form = "global_exchange:select_form"
 
 
 local function mk_select_formspec(p_name, start_i, pagenum)
-	pagenum = math.floor(pagenum)
-	local pagemax = math.floor((selectable_inventory_size - 1) / (8 * 4) + 1)
```

Comment:
Instead of setting selectable_inventory_size and calculating pagemax from it,
pagemax is calculated directly.
https://github.com/HybridDog/global_exchange/commit/7fbd63a180d1bf6224134d7fee31cbe8e9a77bee#commitcomment-16519825